File: Data Structures & Algorithms/cheapest-flight-path/submission-0.py
class Solution:
    def findCheapestPrice(self, n: int, flights: List[List[int]], src: int, dst: int, k: int) -> int:
        prices = [float("inf")] * n
        prices[src] = 0

        for i in range(k+1):
            tempPrices = prices.copy()

            for source, dest, cost in flights:
                if prices[source] != float("inf"): 
                    if prices[source] + cost < tempPrices[dest]:
                        tempPrices[dest] = prices[source] + cost
            prices = tempPrices

        if prices[dst] == float("inf"): return -1
        else: return prices[dst]

                

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int bought = -1;
        int profit = 0;
        for (int i = 0; i < prices.size()-1; i++) {
            if (bought == -1) {
                if (prices[i] < prices[i+1]) {
                    bought = i;
                }
            } else {
                if (prices[i] > prices[i+1]) {
                    profit += (prices[i] - prices[bought]);
                    bought = -1;
                }
            }
        }

        if (bought != -1) {
            profit += prices[prices.size()-1] - prices[bought];
        }

        return profit;
    }
};